server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -829,17 +829,19 @@ private void sync(final Consumer<IndexShard> sync, final String source) {
     }
 
     abstract static class BaseAsyncTask extends AbstractAsyncTask {
+
         protected final IndexService indexService;
 
-        BaseAsyncTask(IndexService indexService, TimeValue interval) {
+        BaseAsyncTask(final IndexService indexService, final TimeValue interval) {
             super(indexService.logger, indexService.threadPool, interval, true);
             this.indexService = indexService;
             rescheduleIfNecessary();
         }
 
         protected boolean mustReschedule() {
-            // don't re-schedule if its closed or if we don't have a single shard here..., we are done
-            return indexService.closed.get() == false;
+            // don't re-schedule if the IndexService instance is closed or if the index is closed
+            return indexService.closed.get() == false
+                && indexService.indexSettings.getIndexMetaData().getState() == IndexMetaData.State.OPEN;
```

Comment:
Note that this will also prevent `AsyncRetentionLeaseSyncTask` tasks to be executed on closed indices shards

Reply:
Can you add a bullet point to the meta-issue that we have to investigate this behavior once the implementation of retention leases is stabilized?

Reply:
Bullet point added